src/components/js/EcApplication.js

```diff
@@ -0,0 +1,77 @@
+import VueMarkdown from 'vue-markdown'
+import EcomApps from '@ecomplus/apps-manager'
+
+export default {
+  name: 'EcApplication',
+  components: {
+    VueMarkdown
+  },
+  data () {
+    return {
+      adminSettings: {},
+      applicationObjId: null,
+      showSettings: false,
+      loading: false,
+      data: null,
+      hidden_data: null
+    }
+  },
+  props: {
+    ecomApps: {
+      type: Object,
+      default: () => new EcomApps()
+    },
+    application: {
+      type: Object,
+      default: () => { }
+    },
+    isInstalled: {
+      type: Boolean,
+      default: false
+    }
+  },
+  methods: {
+    installApp () {
+      this.$message.loading('Instalando aplicativo ' + this.title, 1)
+      this.ecomApps.installApp(this.appId, true)
+        .then(install => {
+          this.$message.success(this.title + ' instalado', 2)
+          this.isInstalled = true
+        })
+        .catch(e => {
+          this.$message.error('Não foi possível instalar o aplicativo', 3)
+        })
+    },
+    deleteApp () {
+      this.ecomApps.removeApplication(this.applicationObjId).then(() => {
```

Comment:
Você está usando `applicationObjId` aqui, mas setou ele em algum lugar?

Reply:
Na verdade eu botei aí porque iria verificar se ele veio como parametro na URL ou então pretendo setá-lo após a instalação.

Reply:
Neste caso deveria ser uma prop, não?
Não é bom bater o router diretamente no source do componente (adiciona contexto).

Reply:
Na verdade eu acho que de qualquer forma não precisa disto, você pega o `_id` dentro do application, seria interessante inclusive você ter um método no componente que busca o body do application se ele não vier completo, e depois de buscar/instalar/deletar/atualizar você emite eventos `this.$emit('update:application', localApplication)`